js/test/test.js

```diff
@@ -348,78 +397,114 @@ async function runPrivateTests(exchange, symbol) {
+        'fetchPositions': [exchange, symbol],
+        'fetchLedger': [exchange, code],
+        'fetchTransactions': [exchange, code],
+        'fetchDeposits': [exchange, code],
+        'fetchWithdrawals': [exchange, code],
+        'fetchBorrowRates': [exchange, code],
+        'fetchBorrowRate': [exchange, code],
+        'fetchBorrowInterest': [exchange, code],
+        'fetchBorrowInterest': [exchange, code, symbol],
+        'addMargin': [exchange, symbol],
+        'reduceMargin': [exchange, symbol],
+        'setMargin': [exchange, symbol],
+        'setMarginMode': [exchange, symbol],
+        'setPositionMode': [exchange, symbol],
+        'setLeverage': [exchange, symbol],
+        'cancelAllOrders': [exchange, symbol],
+        'cancelOrder': [exchange, symbol],
+        'cancelOrders': [exchange, symbol],
+        'fetchCanceledOrders': [exchange, symbol],
+        'fetchClosedOrder': [exchange, symbol],
+        'fetchOpenOrder': [exchange, symbol],
+        'fetchOrder': [exchange, symbol],
+        'fetchOrderTrades': [exchange, symbol],
+        'fetchPosition': [exchange, symbol],
+        'fetchFundingHistory': [exchange, symbol],
+        'fetchDeposit': [exchange, code],
+        'createDepositAddress': [exchange, code],
+        'fetchDepositAddress': [exchange, code],
+        'fetchDepositAddresses': [exchange, code],
+        'fetchDepositAddressesByNetwork': [exchange, code],
+        'editOrder': [exchange, symbol],
+        'fetchBorrowRateHistory': [exchange, symbol],
+        'fetchBorrowRatesPerSymbol': [exchange, symbol],
+        'fetchLedgerEntry': [exchange, code],
+        'fetchPositionsRisk': [exchange, symbol],
+        'fetchWithdrawal': [exchange, code],
+        'transfer': [exchange, code],
+        'withdraw': [exchange, code],
+    };
+
+    const testNames = Object.keys (tests);
+    const errors = [];
+    for (let i = 0; i < testNames.length; i++) {
+        const testName = testNames[i];
+        const testArgs = tests[testName];
+        const success = await testSafe (testName, ...testArgs);
+        if (!success) {
+            errors.push (testName);
+        }
     }
 
-    await test ('fetchTransactions', exchange, code);
-    await test ('fetchDeposits', exchange, code);
-    await test ('fetchWithdrawals', exchange, code);
-    await test ('fetchBorrowRate', exchange, code);
-    await test ('fetchBorrowRates', exchange);
-    await test ('fetchBorrowInterest', exchange, code);
-    await test ('fetchBorrowInterest', exchange, code, symbol);
-
-    if (exchange.extendedTest) {
-
-        await test ('InvalidNonce', exchange, symbol);
-        await test ('OrderNotFound', exchange, symbol);
-        await test ('InvalidOrder', exchange, symbol);
-        await test ('InsufficientFunds', exchange, symbol, balance); // danger zone - won't execute with non-empty balance
+    if (errors.length > 0) {
+        throw new Error ('Failed private tests: ' + errors.join (', '));
```

Comment:
@ttodua I was suggesting something like this ⬆️

Reply:
thats good way and iirc, I had such list in initial version of PR (the linked one in the top comment, around september) but that PR become too intrusive change and it wasn't going to get reviewed, then i've just edited the existing test.js file slightly, so the 'review' become easier for @lead. 
but if you @lead will review such changes, I'll for sure do that way (already had something similar in old commits)

Reply:
if you feel that you can take this PR and submit your further changes, would be good. all i want is to make merge happen faster, however you can.

Reply:
or just give me assignment directly as you think (just describe) and i'll do that within 24 hrs.

Reply:
@ttodua for now, we can only apply this logic to private tests to avoid messing with the public ones (since they're directly tied to the build success). If so, what is missing besides applying the same logic to `extendedTests`?

Reply:
to me, for now, nothing more is remaining. my main goal here is to make this file in condition to become transpilable. I had skeletons for py/php iirc, if this PR is ready for merge, I'll push next things tomorrow.

Reply:
@ttodua do you think this https://github.com/carlosmiei/ast-transpiler could make the transpilation easier?

Reply:
your transpiler would take things on upper level, but at this moment, how/what should we do?

Reply:
i mean, it will be a huge change for transpilation process, and my current intentions were just to comply with our regex transpiler (before we merge your ast-transpiler, it will need some time)